### PR TITLE
skawarePackages: add man page documentation ports

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

[flexibeast](https://github.com/flexibeast) has recently ported some of the skaware documentation to mdoc, meaning we can provide man pages for some of `skawarePackages`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build s6-man-pages s6-networking-man-pages execline-man-pages

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build s6-man-pages s6-networking-man-pages execline-man-pages

File: pkgs/build-support/skaware/build-skaware-man-pages.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  inherit pname version src;
+
+  makeFlags = [
+    "MANPATH=${manDir}"
+  ];
+
+  dontBuild = true;
+
+  preInstall = lib.concatMapStringsSep "\n"
+    (section: ''mkdir -p "${manDir}/man${builtins.toString section}"'')
```
Comment (SuperSandro2000) [inline pkgs/build-support/skaware/build-skaware-man-pages.nix:43]:
```suggestion
    (section: "mkdir -p \"${manDir}/man${builtins.toString section}\"")
```

File: pkgs/build-support/skaware/build-skaware-man-pages.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+{
+  # : string
```
Comment (SuperSandro2000) [inline pkgs/build-support/skaware/build-skaware-man-pages.nix:4]: What are those comments?

Comment (sternenseemann) [reply pkgs/build-support/skaware/build-skaware-man-pages.nix:4]: type hints like `skawarePackages.buildPackage` has.

File: pkgs/build-support/skaware/build-skaware-man-pages.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/build-support/skaware/build-skaware-man-pages.nix:43]: What's the problem here? I thought you found escape sequences in double quoted strings unreadable?

Comment (SuperSandro2000) [reply pkgs/build-support/skaware/build-skaware-man-pages.nix:43]: A while ago there was a PR that removed '' '' from single line strings. Just following that.

File: pkgs/build-support/skaware/build-skaware-man-pages.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/build-support/skaware/build-skaware-man-pages.nix:4]: Never saw those before.

File: pkgs/build-support/skaware/build-skaware-man-pages.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/build-support/skaware/build-skaware-man-pages.nix]: Makes sense. Kinda slipped my mind.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 113118` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>execline-man-pages (skawarePackages.execline-man-pages)</li>
<li>s6-man-pages (skawarePackages.s6-man-pages)</li>
<li>s6-networking-man-pages (skawarePackages.s6-networking-man-pages)</li>
</ul>
</details>

**Comment by Profpatsch** [comment]
Sooo, my problem with this manpage project is that the manpages are not automatically generated from the official upstream docs, but handwritten. So I’m unsure how good they are tbh, and how long they are going to be kept up-to-date with upstream releases. 
e.g. for the last upstream release it took quite a while for them to be updated. In that case, what should we do, distribute out-of-date manpages?

**Comment by sternenseemann** [comment]
For the last release, it was pretty quick, quicker than nixpkgs master updating, in fact :)

If we really worry about them being up to date, we could tie them into their respective packages' version and `throw` or mark them as broken automatically.

Also: they are separate packages and I think this is reasonably clear to the user, it is not really our responsibility to assess how well something will be maintained in the future (or at least we can't really achieve that anyways).

**Comment by Profpatsch** [comment]
They seem to be update quite frequently, I think it’s okay if we add them to the “official” skaware release outputs instead of putting them in different attributes. They might be slightly out of date, but better than no manpage by default.

Can we just pull them in one `skawarePackages.manpages` and then add them to the post-install of build-skaware package?

**Comment by alyssais** [comment]
I don't have a strong opinion, but I think my preference would be to have these as separate packages, since they are separately maintained.

@Profpatsch do you think we could merge this as is, since it's pretty much ready to go with them as seperate packages (and would just need the versions bumped)?

**Comment by sternenseemann** [comment]
> I don't have a strong opinion, but I think my preference would be to have these as separate packages, since they are separately maintained.

Would also be my preference. Having separate man page packages also has a precedent in nixpkgs in general (clang-mangpages, …).

**Comment by Profpatsch** [comment]
Well then let’s do that.

**Comment by sternenseemann** [comment]
Rebased, packages should be up to date as far as I can tell.

**Pull request merged**